core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/CollectInformation.scala

```diff
@@ -81,11 +81,58 @@ class CollectInformation(apps: Seq[ApplicationInfo]) extends Logging {
 
   // get read data schema information
   def getDataSourceInfo: Seq[DataSourceProfileResult] = {
-    val filtered = apps.filter(_.dataSourceInfo.size > 0)
-    val allRows = filtered.flatMap { app =>
+    val dataSourceApps = apps.filter(_.dataSourceInfo.size > 0)
+    val sqlAccums = CollectInformation.generateSQLAccums(dataSourceApps)
+    // Metrics to capture from event log to the result
+    val buffer_time: String = "buffer time"
+    val scan_time = "scan time"
+    val data_size = "data size"
+    val decode_time = "GPU decode time"
+
+    // This is to save the metrics which will be extracted while creating the result.
+    case class IoMetrics(
+        var buffer_time: String,
+        var scan_time: String,
+        var data_size: String,
+        var decode_time: String)
+
+    def getIoMetrics(sqlAccums: Seq[SQLAccumProfileResults]): IoMetrics = {
+      val finalRes = IoMetrics("","","","")
+      sqlAccums.map( x => x.name match {
+        case `buffer_time` => finalRes.buffer_time = x.max_value.toString
+        case `scan_time` => finalRes.scan_time = x.max_value.toString
+        case `data_size` => finalRes.data_size = x.max_value.toString
+        case `decode_time` => finalRes.decode_time = x.max_value.toString
+      })
+      finalRes
+    }
+
+    val allRows = dataSourceApps.flatMap { app =>
+      val appSqlAccums = sqlAccums.filter(sqlAccum => sqlAccum.appIndex == app.index)
+
+      // Filter appSqlAccums to get only required metrics
+      val dataSourceMetrics = appSqlAccums.filter(sqlAccum => sqlAccum.name.contains(buffer_time)
+          || sqlAccum.name.contains(scan_time) || sqlAccum.name.contains(decode_time)
+          || sqlAccum.name.equals(data_size))
+
       app.dataSourceInfo.map { ds =>
-        DataSourceProfileResult(app.index, ds.sqlID, ds.format, ds.location,
-          ds.pushedFilters, ds.schema)
+        if (app.gpuMode) {
```

Comment:
scan time should be in CPU data sources as well, like parquet reads, why are we differentiating gpu vs non gpu vs just check if they are there?

Reply:
Removed this check entirely which is not required. It was the same code in else part.